documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,174 @@
+
+> Authentication Context Class Reference
+>>    OPs MUST support requests for specific Authentication Context Class Reference values via the acr_values parameter, as defined in Section 3.1.2. (Note that the minimum level of support required for this parameter is simply to have its use not result in an error.)
+
+OIDC also defines that the parameter acr_values is OPTIONAL and does not specify possible values. This leads to interoperability issues.
+
+This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
+
+This document defines that Camara Clients SHOULD not use the acr_values parameter. 
+
+> To foster interoperability a future version of this document might define values for the acr_values parameter acceptable in Camara.
+
+## Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects.
+
+### Purpose as part of scope
+
+To improve interoperablitiy this document RECOMMENDS that API Working groups define scope values following this scheme:
+
+`dpv:<dpvValue>#<technical_scope>`
+
+- `<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+- `<technical_scope>` some technical scope needed by the Camara API
+
+This document does not change OIDC definitions of scope values.
+
+
+## Missing "openid" scope
+
+[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
+> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+
+This document defines the following error handling for a missing "openid" value in scope. 
+Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+
+If "openid" is missing in the scope value, the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
+
+## ID Token sub claim
+
+The subject identifier must be globally unique identifier and a case-sensitive ASCII string with a maximum length of 255. The sub value MUST not contain the MSISDN. Within Camara this is populated by a Pseudonymous Customer Reference (PCR) which is a system-facing unique identifier that links a User’s MSISDN/IP-Address to a ASP. It is generated by the OP.
```

Comment:
If you are going to insist that the subject identifier be a "globally unique identifier" called the "Pseudonymous Customer Reference" then you need to specify how to compute that. For otherwise, how can any API provider be sure that the PCR they have generated will not also be generated by a different provider for a different end user?

Also, it is very important to note that an "end user" in the context of a CAMARA API may not have an MSISDN, and is unlikely to have a static IP address. So the PCR calculation method needs to take into account all the end user identifiers that an API provider may be using.

But I think it would be easier simply to require that the subject identifier be locally unique to a given API provider.

Reply:
I agree with being against a globally unique id because that enables user tracking and cross-domain user profile building.

OIDC has PPID
```text
Pairwise Pseudonymous Identifier (PPID)
    Identifier that identifies the Entity to a Relying Party that cannot be correlated with the Entity's PPID at another Relying Party. 
```

And also
```
17.3.  Correlation

To protect the End-User from possible correlation among Clients, the use of a Pairwise Pseudonymous Identifier (PPID) as the sub (subject) SHOULD be considered. 
```

And also [Pairwise Identifier Algorithm](https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg)

If parties want/need to share PPID they can use the above PairwiseAlg and one sector identifier.

Not sure whether this is a Camara concern.

I am going to change this section now

Reply:
Amended ID Token sub claim

Reply:
I think that right now we will not be able to support pairwise identifiers with sector_identifier_uri because this field is not included in the current version of the TMForum Operate API (v5) for client provisioning.

Reply:
> I think that right now we will not be able to support pairwise identifiers with `sector_identifier_uri` because this field is not included in the current version of the TMForum Operate API (v5) for client provisioning.

This profile does not mandate PPID. @garciasolero is the current text OK for you?

Reply:
The current text of the profile says: _This document defines that the sub claim **MUST** be a Pairwise Pseudonymous Identifier (PPID)_. Therefore, it requires the use of PPID.

Reply:
Sorry, I meant does not mandate an certain PPID algorithm. PPID per se should not be a problem, right?
PPID means that two different client ids get different sub values preventing them from using sub for correlation even if the two clients collude.

Reply:
We agree that identifiers should preferably be pairwise. But implementing them is not a straightforward task in those existing solutions that use public identifiers. It could potentially cause issues with existing integrations. Therefore, in those cases, having support for that functionality is unlikely to be available in the short term.